### ISSUE TITLE
Feature Request - Rates Management

### ISSUE THREAD

**Issue by slango-im** [iNextrix/ASTPP#399]
Hello!

How can we create an origination list, which is a list of all our lowest termination rates + markup?

For example, we have 10 termination partners. All 10 have code +38093 with different prices.

But when we want to re-sell, we need to create an A-Z list with only one code of +38093.

The list should have only one code, based on some logic. Which would be the lowest cost + markup.

Is this possible?

**Comment by powerpbx** [comment]
That almost sounds like you are asking for a call simulator.  I know call simulators for one number are fairly common on platforms like this.  So you basically enter the number and it will show the details such as what plan/rate/route it would take etc.  It sounds like you want to do that for all your codes which may be a rather heavy load and may take some time.

I would like to at least see a call simulator for testing individual numbers.

**Comment by slango-im** [comment]
Hi @powerpbx , I'm not sure it's a call simulator. Simply, how can we create our own A-Z list so other terminators can buy from us?

I have 10 termination partners. I want to take the best price of each one and create an A-Z origination list to give to buyers.

... Does that make sense?

**Comment by devangn** [comment]
@slango-im As per existing configuration, you can create a list of origination rates manually for particular prefix and set the force-trunk to whichever trunk under one rate group and assign that rate group to whoever you want sell that rate card.
For automatic procedure, you need quite a modification as per your exact requirement. If you need we may help you with you it, contact us sales@inextrix.com

**Comment by AlexHeylin** [comment]
In mysql, something like this
```sql
select comment, replace(replace(pattern,".*",""),"^","") as prefix, min(cost) as MinPrice from routes
group by pattern
order by comment ASC
```
I'm not sure this is the best way to do this, but it seems to work. Test yourself before relying on this.

Note this will only match exact prefix / pattern / code matching, it will not take account of longer or shorter prefixes which may be cheaper. However this is what you requested.

You requested this was based on termination rates + markup, however this uses the origination rate card cost instead. It should give you an idea where to head.

**Comment by Hemdip** [comment]
@slango-im that feature is available in V-4.0 you can simply duplicate rategroup and it will copy all origination rates also. You can update version and use that feature
